core/sql/executor/ExHdfsScan.h

```diff
@@ -429,41 +432,92 @@ class ExOrcFastAggrTcb  : public ExOrcScanTcb
 
 #define RANGE_DELIMITER '\002'
 
-inline char *hdfs_strchr(const char *s, int c, const char *end, NABoolean checkRangeDelimiter)
+inline char *hdfs_strchr(char *s, int c, const char *end, NABoolean checkRangeDelimiter, int mode = 0)
 {
   char *curr = (char *)s;
-
-  while (curr < end) {
+  int count=0;
+  if( (mode & HIVE_MODE_DOSFORMAT ) == 0)
+  {
+   while (curr < end) {
     if (*curr == c)
+    {
        return curr;
+    }
     if (checkRangeDelimiter &&*curr == RANGE_DELIMITER)
        return NULL;
     curr++;
+   }
+  }
+  else
+  {
+   while (curr < end) {
+     if (*curr == c)
+     {
+         if(count>0 && c == '\n')
+         {
+           if(s[count-1] == '\r') s[count-1] = ' '; 
```

Comment:
I tested with DATE and INT, both works well with extra space. But maybe \0 is better. Let me try it out.